src/server/pfs/server/driver_fsck.go

```diff
@@ -377,18 +376,23 @@ func (d *driver) fsck(ctx context.Context, fix bool, cb func(*pfs.FsckResponse)
 	}
 	for _, repo := range repoInfos {
 		if err := dbutil.WithTx(ctx, d.env.DB, func(ctx context.Context, tx *pachsql.Tx) error {
-			commitInfo := &pfs.CommitInfo{}
-			if err := d.commits.ReadWrite(tx).GetByIndex(pfsdb.CommitsRepoIndex, pfsdb.RepoKey(repo.Repo), commitInfo, col.DefaultOptions(), func(string) error {
-				commitInfos[pfsdb.CommitKey(commitInfo.Commit)] = proto.Clone(commitInfo).(*pfs.CommitInfo)
-				return nil
-			}); err != nil {
+			commits, err := pfsdb.ListCommitTxByFilter(ctx, tx, pfsdb.CommitListFilter{pfsdb.CommitRepos: []string{pfsdb.RepoKey(repo.Repo)}}, false, false)
+			if err != nil {
 				return errors.Wrap(err, "get commits by commits repo index")
 			}
-			branchInfo := &pfs.BranchInfo{}
-			err := d.branches.ReadWrite(tx).GetByIndex(pfsdb.BranchesRepoIndex, pfsdb.RepoKey(repo.Repo), branchInfo, col.DefaultOptions(), func(string) error {
-				branchInfos[pfsdb.BranchKey(branchInfo.Branch)] = proto.Clone(branchInfo).(*pfs.BranchInfo)
+			for _, commitInfo := range commits {
+				commitInfos[pfsdb.CommitKey(commitInfo.Commit)] = commitInfo
+			}
+			iter, err := pfsdb.NewBranchIterator(ctx, tx, 0, 100, &pfs.Branch{Repo: repo.Repo})
+			if err != nil {
+				return errors.Wrap(err, "delete repo info")
+			}
+			if err := stream.ForEach[pfsdb.BranchInfoWithID](ctx, iter, func(branchInfoWithID pfsdb.BranchInfoWithID) error {
+				branchInfos[pfsdb.BranchKey(branchInfoWithID.Branch)] = branchInfoWithID.BranchInfo
```

Comment:
Writing outside the transaction is tricky.  This is okay because it's a full overwrite.  Appending here would not be okay.

Reply:
I think this is only read. The stream is iterating through the branches, and we save each branch in the local map.